src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            v, f = trimesh.util.append_faces([i[0] for i in vf], [i[1] for i in vf])
+            polygon_mesh = trimesh.creation.extrude_triangulation(vertices=v, faces=f, height=height)
+
+            # Translate the polygon mesh to the desired height.
+            polygon_mesh.vertices[:,2] += centerZ - polygon_mesh.bounding_box.center_mass[2]
+
+            # Check if we have a valid volume
+            if polygon_mesh.is_volume:
+                if tol_size is not None and tol_size >= 0.01:
+                    warnings.warn(f"Computing bounded footprint of polygon resulted in invalid volume, "
+                                  f"but was able to remedy this by buffering/simplifying polygon by {tol_size}")
+                break
+
+        if not polygon_mesh.is_volume:
+            raise RuntimeError("Computing bounded footprint of polygon resulted in invalid volume")
+
+        return MeshVolumeRegion(polygon_mesh, centerMesh=False)
+
+    def buffer(self, amount):
+        buffered_polygon = self.polygon.buffer(amount)
+
+        if isinstance(buffered_polygon, shapely.geometry.Polygon):
+            buffered_polygon = MultiPolygon([buffered_polygon])
+
+        return PolygonalFootprintRegion(polygon=buffered_polygon, name=self.name)
+
+    @cached_property
+    def isConvex(self):
+        return self.polygons.equals(self.polygons.convex_hull)
+
+class PathRegion(Region):
+    """A region composed of multiple polylines in 3D space.
+
+    One of points or polylines should be provided. If both are provided,
+    points will be used.
+
+    Args:
+        points: A list of points defining a single polyline.
+        polylines: A list of list of points, defining multiple polylines.
+    """
+    def __init__(self, points=None, polylines=None, name=None):
+        super().__init__(name)
+        # Standardize inputs
+        if points is not None:
+            polylines = [points]
+        elif polylines is not None:
+            pass
+        else:
+            raise ValueError("PathRegion created without specifying points or polylines")
+
+        # Extract a list of vertices and edges, without duplicates
+        self.vec_to_vert = {}
+        self.edges = []
+        vertex_iter = 0
+
+        for pls in polylines:
+            last_pt = None
+
+            for pt in pls:
+                # Extract vertex
+                cast_pt = toVector(pt)
+
+                if cast_pt not in self.vec_to_vert:
+                    self.vec_to_vert[cast_pt] = vertex_iter
+                    vertex_iter += 1
+
+                # Extract edge, if not first point
+                if last_pt is not None:
+                    new_edge = self.vec_to_vert[last_pt], self.vec_to_vert[cast_pt]
+                    self.edges.append(new_edge)
+
+                # Update last point
+                last_pt = cast_pt
+
+        self.vert_to_vec = {val:key for key,val in self.vec_to_vert.items()}
```

Comment:
Might as well make this a tuple or numpy array and not bother with a hash table.